java/arcs/core/host/Plan.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+/**
+ * A [Plan] is usually produced by running the build time Particle Accelerator tool, it consists
+ * of a set of specs for handles, particles used in a recipe, and mappings between them.
+ */
+open class Plan(
+    val handleSpecs: List<HandleSpec>,
+    val particleSpecs: List<ParticleSpec>,
+    val handleConnectionSpecs: List<HandleConnectionSpec>
```

Comment:
handleConnectionSpec contain the corresponding Particle and Handle, so the list of particles and the list of handles can be inferred from the connections. is there really a need to pass all three?

Reply:
Yeah I thought about that, I could change it to be that way, but then I thought to list the graph in Vertex/Edge format. I should probably change it back, since there's a slight chance that you could have duplicate HandleSpec for the same handle, especially after Parceling them on Android if not careful when deserializing.